python/tank/platform/engine.py

```diff
@@ -970,6 +970,8 @@ def register_command(self, name, callback, properties=None):
                 # we know the app for the existing item.
                 # so prefix with app name
                 prefix = existing_item["properties"].get("app").instance_name
+                if existing_item["properties"].get("group"):
+                    prefix = "%s:%s" % (prefix, existing_item["properties"]["group"])
```

Comment:
In the following test i register the following two sets of properties:

```
{'description': 'Launches and initializes an application environment.',
 'group': 'Group 1',
 'group_default': False,
 'icon': '/mnt/wintermute/the_evil_mimes/install/app_store/tk-nuke/v0.6.9/icon_256.png',
 'short_name': 'launch_nuke',
 'title': 'Launch Nuke'}
```
and
```
{'description': 'Launches and initializes an application environment.',
 'group': 'Group 2',
 'group_default': False,
 'icon': '/mnt/wintermute/the_evil_mimes/install/app_store/tk-nuke/v0.6.9/icon_256.png',
 'short_name': 'launch_nuke',
 'title': 'Launch Nuke'}
```

If I print `engine.commands`, I get:

```
'tk-multi-launchnuke:Group 1:launch_nuke': {'callback': <function callback_wrapper at 0x109718b18>,
                                             'properties': {'app': <Sgtk App 0x10972a190: tk-multi-launchapp, engine: <Sgtk Engine 0x10339f050: tk-shell, env: project>>,
                                                            'description': 'Launches and initializes an application environment.',
                                                            'group': 'Group 1',
                                                            'group_default': False,
                                                            'icon': '/mnt/wintermute/the_evil_mimes/install/app_store/tk-nuke/v0.6.9/icon_256.png',
                                                            'prefix': 'tk-multi-launchnuke:Group 1',
                                                            'short_name': 'launch_nuke',
                                                            'title': 'Launch Nuke'}},
 'tk-multi-launchnuke:launch_nuke': {'callback': <function callback_wrapper at 0x109718cf8>,
                                     'properties': {'app': <Sgtk App 0x10972a190: tk-multi-launchapp, engine: <Sgtk Engine 0x10339f050: tk-shell, env: project>>,
                                                    'description': 'Launches and initializes an application environment.',
                                                    'group': 'Group 2',
                                                    'group_default': False,
                                                    'icon': '/mnt/wintermute/the_evil_mimes/install/app_store/tk-nuke/v0.6.9/icon_256.png',
                                                    'prefix': 'tk-multi-launchnuke',
                                                    'short_name': 'launch_nuke',
                                                    'title': 'Launch Nuke'}},
```

This is inconsistent - i would expect consistency in the way all colliding names are named. So i would expect the following two keys to show up whenever a group collision is detected:

- `'tk-multi-launchnuke:Group 1:launch_nuke'`
- `'tk-multi-launchnuke:Group 2:launch_nuke'`